[v8.5.x] AngularSupport: Update description for angular_support_enabled config option (#49573)
grafana/grafana: docs/sources/developers/angular_deprecation.md

# Angular support deprecation

> Angular plugin support is deprecated, and it will be removed in a future release. There are still many community plugins that depend on Grafana’s angular plugin support for them to work. The same is true for many internal (private) plugins developed over the years by Grafana users. Grafana version 9 will have a server configuration option, global for the whole instance, that will control if angular plugin support is available or not. By default angular support is still enabled but this will soon change as we complete the migration of all angular code in the core product.

Reviewer: ```suggestion
Angular plugin support is deprecated and will be removed in a future release. There are still many community plugins that depend on Grafana’s angular plugin support for them to work. The same is true for many internal (private) plugins developed over the years by Grafana users. Grafana version 9 will have a server configuration option, global for the whole instance, that will control whether angular plugin support is available or not. By default, angular support will still be enabled.  This will change as we complete the migration of all angular code in the core product.
```

## Why are we deprecating angular support?

AngularJS is an old frontend framework that stopped active development many years ago. As a result, it is a security risk. AngularJS also requires unsafe-eval in the CSP (Content Security Policy) settings which also reduces the security level of how javascript is executed in the browser.

## When will angular plugins stop working?

> We hope to migrate the last remaining angular code in core Grafana later this year. Then the setting enable_angular_support can be disable by default. This will make all angular plugins stop working. If you still depend on community or internally developed plugins that require AngularJS then you will have to enable this option.

Reviewer: ```suggestion
We hope to migrate the last remaining angular code in core Grafana later this year. You will then be able to set  enable_angular_support to disabled by default. This will make all angular plugins stop working. If you still depend on community or internally developed plugins that require AngularJS,  you will have to enable this option.
```
